### ISSUE TITLE
Authorization problem in accessing a mutation resolver from a lambda function resolver

### ISSUE THREAD

**Issue by npapadak** [aws-amplify/amplify-category-api#2106]
Hi,
I have a type model named Order and a type Mutation function resolver named processOrder, defined in my schema.graphql assigning them the authorization policies bellow:

```
type Order
  @model
  @auth(
    rules: [
      { allow: owner, operations: [read, create] }
      { allow: groups, groups: ["Admin"] }
      { allow: public, operations: [create], provider: iam }
    ]
  ) {
  order: [AWSJSON]
  price: Float
  served: Boolean
  tableNumber: Int
  stopwatchIsRunning: Boolean
  orderNumber: Int
}

type Mutation {
  processOrder(input: OrderInput): String
    @function(name: "saveOrder-${env}")
    @auth (
    rules: [
      { allow: public, provider: iam }
      { allow: private }
    ]
  )
}
```

It's multi-auth and while authorised users, managed by cognito (the default authorization mode) work fine, the guest users that are managed by IAM cannot access the auto-generated by appsync, createOrder mutation, which is called in the lambda function saveOrder above (@function(name: "saveOrder-${env}")).

I call the processOrder mutation from a react frontend as follows with no issues:

```
const proccessOrder = async (input, user) => {
  try {
    await API.graphql({
      ...graphqlOperation(processOrder, { input: input}),
      authMode: !user ? 'AWS_IAM' : 'AMAZON_COGNITO_USER_POOLS'
    })
    .then((res) => console.log('Checkout response: ',res.data.processOrder))  
  } catch (err) {
    console.log("Checkout error: ", err)
  }
}
```

And that gets me to my saveOrder lambda where I initialize the AWS Appsync Client depending on whether dealing with authenticated or guest users as follows:

```
const initializeClient = () => {
  let auth
  if(token.includes('Credential')){
    auth = {
      type: AUTH_TYPE.AWS_IAM,
      credentials: credentials 
    }
  } else {
    auth = {
      type: AUTH_TYPE.AMAZON_COGNITO_USER_POOLS,
      jwtToken: token
    }
  }
  
  let client = new AWSAppSyncClient({
    url: appsyncUrl,
    region,
    auth: auth,
    disableOffline: true
  })

  return client
} 
```

And I finally call the mutation createOrder from the same lambda function (@function(name: "saveOrder-${env}")) as follows:

```
const saveOrder = async (totalOrder) => {
    try {
      const result = await initializeClient().mutate({
        mutation: gql(createOrder), 
        variables: {
          input: totalOrder
        },
      })
      console.log("saveOrder response: ", result)
      return {
        statusCode: 200,
        body: JSON.stringify(result)
      }
    } catch (error) {
      console.log("saveOrder/saveOrder error: ", JSON.stringify(error) )
      return error
    }
  }
```

Everything cognito wise (meaning authorised users) work fine and can access the createOrder mutation above. But guest users cannot.

I've also activated the guest access in my Cognito Identity pool, assigning the following super generous policy to its role:

```
{
    "Sid": "VisualEditor0",
    "Effect": "Allow",
    "Action": "appsync:*",
    "Resource": [
        "arn:aws:appsync:'region':'ACCOUNT#':apis/'API-ID'/*",
    ]
}
```

And I've also assigned the super generous policy to the saveOrder lambda function role as follows:

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Action": [
                "appsync:*"
            ],
            "Resource": "*",
            "Effect": "Allow"
        }
    ]
}
```
When checking the above role in IAM Policy Simulator, it seems like it has the necessary permissions to access the createOrder mutation as expected. 

Finally, checking on the Schema at the AWS console, it seems to comply with everything defined in schema.graphql as you can see below:
```
type Order @aws_iam
@aws_cognito_user_pools {
	order: [AWSJSON]
	price: Float
	served: Boolean
	tableNumber: Int
	stopwatchIsRunning: Boolean
	orderNumber: Int
	id: ID!
	createdAt: AWSDateTime!
	updatedAt: AWSDateTime!
	owner: String
}

type Mutation {
	updateOrder(input: UpdateOrderInput!, condition: ModelOrderConditionInput): Order
	deleteOrder(input: DeleteOrderInput!, condition: ModelOrderConditionInput): Order
	createOrder(input: CreateOrderInput!, condition: ModelOrderConditionInput): Order
		@aws_iam
@aws_cognito_user_pools
	processOrder(input: OrderInput): String
		@aws_iam
@aws_cognito_user_pools
}
```

As I said, everything cognito wise works fine, meaning that it's not a code issue. Problem is with guest users and IAM, with cloudwatch logs keep on returning the following message, tho my super generous policies:

```
"networkError": null,
"message": "GraphQL error: Not Authorized to access createOrder on type Mutation"
```

I have created my lambda function using amplify cli and the command:
amplify add function 

Amplify version: 12.7.0

What am I missing out?

Thanks a ton!!!

**Comment by DevTGhosh** [comment]
I am having the same issue. I have a graphql lambda resolver that access another grapqhl lambda resolver through graphql api and it has been showing me the unauthorization issue for the past week. Everything was working fine till last week. No changes have been made from my side that could affect this.

**Comment by phani-srikar** [comment]
This seems to be related to https://github.com/aws-amplify/amplify-category-api/issues/2060. This regression is fixed in https://github.com/aws-amplify/amplify-category-api/pull/2062 and pending release

**Comment by npapadak** [comment]
> This seems to be related to #2060. This regression is fixed in #2062 and pending release

Thank you @phani-srikar

**Comment by dpilch** [comment]
The fix is available in v12.10.0 of the CLI.
